driver/src/test/scala/org/mongodb/scala/internal/ScalaObservableSpec.scala

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mongodb.scala.internal
+
+import scala.util.{ Failure, Success }
+
+import com.mongodb.MongoException
+
+import org.mongodb.scala._
+import org.scalatest.{ FlatSpec, Matchers }
+
+class ScalaObservableSpec extends FlatSpec with Matchers {
+
+  "ScalaObservable" should "allow for inline subscription" in {
+    var counter = 0
+    observable().subscribe((res: Int) => counter += 1)
+    counter should equal(100)
+
+    var thrown = false
+    observable(fail = true).subscribe((res: Int) => (), (t: Throwable) => thrown = true)
+    thrown should equal(true)
+
+    var completed = false
+    observable().subscribe((res: Int) => (), (t: Throwable) => (), () => completed = true)
+    completed should equal(true)
+  }
+
+  it should "have a foreach method" in {
+    var counter = 0
+    observable().foreach((res: Int) => counter += 1)
+    counter should equal(100)
+  }
+
+  it should "have a transform method" in {
+    var completed = false
+    var lastSeenString = ""
+    observable[Int]()
+      .transform((res: Int) => res.toString, (ex: Throwable) => ex)
+      .subscribe((s: String) => lastSeenString = s, (t: Throwable) => (), () => completed = true)
```

Comment:
Why not gather a list of all the transformed values and assert that the list contains the expected values, instead of only asserting about the last value seen?